sdk/storage/storage-internal-avro/src/AvroParser.ts

```diff
@@ -0,0 +1,365 @@
+import { AvroReadable } from "./AvroReadable";
+import { KeyValuePair } from "./utils/utils.common";
+import { AvroConstants } from "./AvroConstants";
+
+export class AvroParser {
+  /**
+   * Reads a fixed number of bytes from the stream.
+   *
+   * @static
+   * @param stream
+   * @param length
+   */
+  public static async readFixedBytes(stream: AvroReadable, length: number): Promise<Uint8Array> {
+    const bytes = await stream.read(length);
+    if (bytes.length != length) {
+      throw new Error("Hit stream end.");
+    }
+    return bytes;
+  }
+
+  /**
+   * Reads a single byte from the stream.
+   *
+   * @static
+   * @param stream
+   */
+  private static async readByte(stream: AvroReadable): Promise<number> {
+    const buf = await AvroParser.readFixedBytes(stream, 1);
+    return buf[0];
+  }
+
+  private static async readZigZagLong(stream: AvroReadable): Promise<number> {
+    // copied from https://github.com/apache/avro/blob/master/lang/js/lib/utils.js#L321
```

Comment:
Apache avor follows Apache License 2.0, which requires copyright information when re-distribute it's code.

We need to follow procedures to add dependency for 3rd party source code.

You can re-implement the algorithm instead directly copy.

Reply:
Then what about the MIT licensed one:
https://github.com/mtth/avsc/blob/master/lib/utils.js#L470